meta/TreeMasses.m

```diff
@@ -408,6 +409,20 @@
 
 IsElectricallyCharged[par_] := GetElectricCharge[par] != 0;
 
+IsChargedUnder[field_, vector_] := Block[{},
+  (*First of all, make sure we have been passed a vector*)
+  If[IsVector[vector],
+    (*Check 2 special cases first which are quicker*)
+    Which[IsPhoton[vector], Return[IsElectricallyCharged[field]];,
+     IsGluon[vector], Return[ColorChargedQ[field]];,
+     (*Else check that this field coupled with its anti-field can emit this vector*)
+     (*Note this will not work for vectors that couple to two different fields, e.g. W-bosons*)
+     True, Return[SARAH`Vertex[{SARAH`AntiField[field], field, 
+            vector}, UseDependences -> True][[2, 1]] =!= 0];],
+    Return[None];
```

Comment:
Why do you return `None` here? The other branches seem to return `True` or `False`.
I would have thought one should return `False` if it is not a vector.

Reply:
I used None as a sort of 'error' result, as there really should not be anything passed here except a vector.

Reply:
If I understand correctly, the `Block` could simply be omitted and the function could be streamlined to:
~~~.mathematica
IsChargedUnder[field_, vector_] :=
  IsVector[vector] && Which[
    IsPhoton[vector], IsElectricallyCharged[field],
    IsGluon[vector], ColorChargedQ[field],
    True, SARAH`Vertex[...][[2, 1]] =!= 0
  ]
~~~

Reply:
Ok, I see. Do you know whether the function can than still be used in an `If` statement?
Like so:
~~~.mathematica
If[IsChargedUnder[VP,VP],
   (* True branch *),
   (* False branch),
   (* unknown branch *)
]
~~~
Will the `unknown branch` be used if the result is `None`?

Reply:
Yes, the `unknown branch` will be used if the result is `None` specifically if we input `hh` or `Fe` for the vector for example.  The example you wrote above should return `False`, i.e. the photon is electrically neutral.

Reply:
Ok, then I have another proposal. :) 
The error that you're trying to express is that the argument that is passed in as `vector` is not a vector. I.e. it is a requirement that a vector is passed as second argument. Then I think we could better express this in the function signature:
~~~.mathematica
IsChargedUnder[field_, vector_?IsVector] := ...
~~~
Then this function would be genuine predicate that only returns `True` or `False` and nothing else.
What do you think?

Reply:
I like it, it is now implemented.  Thanks!

Reply:
Thanks for this! May I still make a suggestion? :)
You can omit the `Return` statements and simply write:
~~~.mathematica
IsChargedUnder[field_, vector_?IsVector] := 
  Which[(*Check 2 special cases first which are quicker*)
    IsPhoton[vector], IsElectricallyCharged[field],
    IsGluon[vector], ColorChargedQ[field],
    (*Else check that this field coupled with its anti-field can emit this vector*)
    (*Note this will not work for vectors that couple to two different fields, e.g. W-bosons*)
    True, SARAH`Vertex[{SARAH`AntiField[field], field, vector}, UseDependences -> True][[2, 1]] =!= 0
  ]
~~~

Reply:
Done!  All good.